src/DynamoCoreWpf/UI/GuidedTour/Step.cs

```diff
@@ -246,6 +252,92 @@ internal void UpdatePlacementTarget()
                 stepUIPopup.PlacementTarget = ownedWindow;
                 UpdateLocation();
             }
+            //This case will be used for UIElements that are in the Dynamo VisualTree but they are shown until there is a user interaction (like the SideBar cases)
+            var hostUIElement = Guide.FindChild(MainWindow, HostPopupInfo.HostUIElementString);
+            if (hostUIElement == null)
+                return;
+            HostPopupInfo.HostUIElement = hostUIElement;
+            stepUIPopup.PlacementTarget = hostUIElement;
+            UpdateLocation();
+        }
+
+        /// <summary>
+        /// This method will update the CutOff rectangle size everytime that the step change
+        /// </summary>
+        /// <param name="bVisible">It will say it the CutOff Area will be disabled or enable</param>
+        private void SetCutOffSectionSize(bool bVisible)
+        {
+            if (HostPopupInfo.CutOffRectArea == null)
+                return;
+            if(bVisible)
+            {
+                //This will validate that HostPopupInfo.HostUIElement is in the MainWindow VisualTree otherwise the TransformToAncestor() will crash
+                var foundUIElement = Guide.FindChild(MainWindow, HostPopupInfo.HostUIElementString);
+                if (foundUIElement == null)
+                    return;
+
+                Point relativePoint = HostPopupInfo.HostUIElement.TransformToAncestor(MainWindow)
+                              .Transform(new Point(0, 0));
+
+                var holeWidth = HostPopupInfo.HostUIElement.DesiredSize.Width + HostPopupInfo.CutOffRectArea.WidthBoxDelta;
+                var holeHeight = HostPopupInfo.HostUIElement.DesiredSize.Height + HostPopupInfo.CutOffRectArea.HeightBoxDelta;
+
+                if (StepGuideBackground.CutOffBackgroundArea != null)
+                {
+                    StepGuideBackground.CutOffBackgroundArea.CutOffRect = new Rect(relativePoint.X, relativePoint.Y, holeWidth, holeHeight);
+                }
+            }
+            else
+            {
+                StepGuideBackground.ClearCutOffSection();
+            }
+        }
+
+        /// <summary>
+        /// This method will set the highlight rectangle color if there is any configured in the json file
+        /// </summary>
+        /// <param name="bVisible">It will say it the Highlight Area will be disabled or enable</param>
```

Comment:
`if the Highlight Area will be disabled or enable`? Please update

Reply:
@QilongTang fixed in the next commit:
https://github.com/DynamoDS/Dynamo/pull/12277/commits/7b9f128012709949425ccbf45f7bf3f52d19c89a